universal-application-tool-0.0.1/app/services/program/ProgramServiceImpl.java

```diff
@@ -28,15 +30,18 @@
   private final ProgramRepository programRepository;
   private final QuestionService questionService;
   private final HttpExecutionContext httpExecutionContext;
+  private final Provider<VersionRepository> versionRepositoryProvider;
 
   @Inject
   public ProgramServiceImpl(
       ProgramRepository programRepository,
       QuestionService questionService,
+      Provider<VersionRepository> versionRepositoryProvider,
```

Comment:
Question (non-blocking): why this can't be just versionRepository?

Reply:
Oh, probably could!